File: Service/Translators/BudgetTranslator.vb
Imports BO
Imports DAL
Imports System.Text.RegularExpressions
Public Class BudgetTranslator
    Public Shared Function TranslateEntityToBO(_entity As ProjectBudget, bo As BudgetActivityBO) As ErrorCode
        If _entity Is Nothing Then Return ErrorCode.EntityNull
        If bo Is Nothing Then Return ErrorCode.BoNull
        bo.Id = _entity.ID
        bo.ProjectId = _entity.ProjectID
        bo.Price = _entity.Price
        Dim act As New ActivityBO
        bo.Activity = act
        Dim err = ActivityTranslator.TranslateEntityToBO(_entity.Activity, bo.Activity)
        If (err <> ErrorCode.Success) Then Return err
        Return ErrorCode.Success
    End Function

    Friend Shared Function TranslateBOToEntity(_entity As ProjectBudget, bo As BudgetActivityBO, uow As UnitOfWork) As ErrorCode
        If _entity Is Nothing Then Return ErrorCode.EntityNull
        If bo Is Nothing Then Return ErrorCode.BoNull
        _entity.ProjectID = bo.ProjectId
        _entity.ActivityID = bo.Activity.ID
        _entity.Price = bo.Price
        Return ErrorCode.Success
    End Function

End Class

File: ServiceCore/Translators/BudgetTranslator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using BOCore;
using DALCore;
using DALCore.Models;

namespace ServiceCore.Translators
{
    public class BudgetTranslator
    {
        public static ErrorCode TranslateEntityToBO(ProjectBudget _entity, BudgetActivityBO bo)
        {
            if (_entity == null)
                return ErrorCode.EntityNull;
            if (bo == null)
                return ErrorCode.BoNull;
            bo.Id = _entity.Id;
            bo.ProjectId = _entity.ProjectId;
            bo.Price = _entity.Price;
            ActivityBO act = new ActivityBO();
            bo.Activity = act;
            var err = ActivityTranslator.TranslateEntityToBO(_entity.Activity, bo.Activity);
            if ((err != ErrorCode.Success))
                return err;
            return ErrorCode.Success;
        }

        internal static ErrorCode TranslateBOToEntity(ProjectBudget _entity, BudgetActivityBO bo, UnitOfWork uow)
        {
            if (_entity == null)
                return ErrorCode.EntityNull;
            if (bo == null)
                return ErrorCode.BoNull;
            _entity.ProjectId = bo.ProjectId;
            _entity.ActivityId = bo.Activity.ID;
            _entity.Price = bo.Price;
            return ErrorCode.Success;
        }
    }
}
